File: Android/app/src/main/kotlin/jenga/clock/FirebaseTracker.kt
package jenga.clock

import com.google.firebase.Firebase
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.analytics
import com.google.firebase.analytics.logEvent
import skip.lib.Dictionary

class FirebaseTracker: Tracker {
    override fun trackScreenView(screenName: String) {
        Firebase.analytics.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW) {
            param(FirebaseAnalytics.Param.SCREEN_NAME, screenName)
            param(FirebaseAnalytics.Param.SCREEN_CLASS, "MainActivity")
        }
    }

    override fun trackEvent(eventName: String, parameters: Dictionary<String, Any>?) {
        Firebase.analytics.logEvent(eventName) {
            parameters?.forEach { (key, value) ->
                param(key, value.toString())
            }
        }
    }
}

File: Darwin/Sources/FirebaseTracker.swift
//
//  FirebaseTracker.swift
//  Jenga-Clock
//
//  Created by Adam Kopeć on 09/03/2025.
//

import JengaClock
import FirebaseAnalytics

class FirebaseTracker: Tracker {
    func trackScreenView(screenName: String) {
        Analytics.logEvent(AnalyticsEventScreenView,
                           parameters: [
                            AnalyticsParameterScreenName: screenName,
                            AnalyticsParameterScreenClass: screenName
                           ])
    }
    
    func trackEvent(_ eventName: String, parameters: [String : Any]?) {
        Analytics.logEvent(eventName, parameters: parameters)
    }
    
    init(isAppClip: Bool = false) {
        if isAppClip {
            Analytics.setUserProperty("true", forName: "AppClip")
        }
    }
}
